src/network.cpp

```diff
@@ -266,141 +379,173 @@ namespace xiloader
                         continue;
                     else if (ch == '\b')
                     {
-                        if (g_Password.size())
+                        if (globals::g_Password.size())
                         {
-                            g_Password.pop_back();
+                            globals::g_Password.pop_back();
                             std::cout << "\b \b";
                         }
                     }
                     else
                     {
-                        g_Password.push_back(ch);
+                        globals::g_Password.push_back(ch);
                         std::cout << '*';
                     }
                 }
                 std::cout << std::endl;
 
                 char event_code = (input == "1") ? 0x10 : 0x30;
-                sendBuffer[0x20] = event_code;
+
+                if (input == "3")
+                {
+                    std::string confirmed_password = "";
+                    do
+                    {
+                        std::cout << "Enter new password (6-32 characters): ";
+                        confirmed_password = "";
+                        new_password       = "";
+                        std::cin >> new_password;
+                        std::cout << "Repeat Password           : ";
+                        std::cin >> confirmed_password;
+                        std::cout << std::endl;
+
+                        if (new_password != confirmed_password)
+                        {
+                            xiloader::console::output(xiloader::color::error, "Passwords did not match! Please try again.");
+                        }
+                    } while (new_password != confirmed_password);
+                    new_password = confirmed_password;
+                }
+                sendBuffer[0x39] = event_code;
             }
             /* User wants to create a new account.. */
             else if (input == "2")
             {
             create_account:
```

Comment:
TODO for another time: kill all goto and labels

Reply:
Added a comment for TODO